components/support/nimbus-fml/src/defaults/validator.rs

```diff
@@ -222,6 +244,254 @@ impl<'a> DefaultsValidator<'a> {
             }),
         }
     }
+
+    #[allow(clippy::too_many_arguments)]
+    fn validate_string_aliases(
+        &self,
+        path: &str,
+        error_path: &[String],
+        typ: &TypeRef,
+        value: &Value,
+        defn: &HashMap<&str, &PropDef>,
+        skip: &Option<TypeRef>,
+        errors: &mut Vec<FMLError>,
+    ) {
+        // As an optimization (to stop validating the definition against itself),
+        // we want to skip validation on the `skip` type ref: this is only set by the property defining
+        // a string-alias.
+        let should_validate = |v: &TypeRef| -> bool { !matches!(skip, Some(t) if t == v) };
```

Comment:
TypeRef impls Eq, so can't this be `skip != Some(v)`?

Reply:
Oh yes! That's a good idea.

I settled on `skip.as_ref() != Some(v)`.